src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -28,6 +29,43 @@ public NodeSelectorState(Topology topology)
                 UnlikelyEveryoneFaultedChoiceIndex = 0;
             }
 
+            public NodeSelectorState(Topology topology, NodeSelectorState prevState)
+            {
+                Topology = topology;
+                Nodes = topology.Nodes;
+                Failures = new int[topology.Nodes.Count];
+                FastestRecords = new int[topology.Nodes.Count];
+                UnlikelyEveryoneFaultedChoiceIndex = 0;
+                
+                var fastestNode = prevState.Nodes.ElementAt(prevState.Fastest);
```

Comment:
Is there a chance here that prevState.Fastest will cause index out of range? can be -1 for example? you can switch dynamically using server-side that mode, what will happen then?

Reply:
The only places that set `Fastest` have the set value checked for range. No place in the code sets this as `-1`, the default of `Fastest` is `0`. There shouldn't be any out of range errors as long as we are using the `Nodes` and 'Fastest' of the same state object. But just in case, I added a check that will revert to the original behavior (do nothing) if it is out of range